fix: Rehaul docker installation guide (#1031)
appsmithorg/appsmith-docs: website/docs/getting-started/setup/instance-management/update-appsmith.md

* [Manual update](#manual-update)
* [Automatic update](#automatic-update)

### Manual update
To update Appsmith manually, go to the root directory of the installation and run the following command:

```
docker-compose pull && docker-compose rm -fsv appsmith && docker-compose up -d
```

### Automatic update

To update Appsmith automatically, you need to turn on the automatic updates. Follow the below steps:

1. Go to the root directory of the Appsmith installation and run:

   ```
   docker-compose down
   ```

2. Open the `docker-compose.yml` file with any text editor and uncomment lines 13-23.
3. Save the file and run:

   ```
   docker-compose up -d
   ```

The server restart updates the Appsmith to the latest version. You can also choose to schedule automatic updates. For more information, see [Configure a maintenance window for automatic updates](/getting-started/setup/instance-management/maintenance-window#adding-a-configurable-maintenance-window-for-appsmiths-auto-updates).

## Troubleshooting

> The auto backup feature is also turned on when you turn on automatic updates. This means that Appsmith is first backed up before updating to the latest version. If you see any errors during deployment, you can use the backup version to roll back. For more information, see [the rollback instructions using the `appsmithctl` command](/getting-started/setup/instance-management/appsmithctl#restore-appsmith-instance). If you continue to experience issues, please raise them on the <a href="#!" onclick="Intercom('show')">chat window</a> available on the documentation.

Reviewer: ```suggestion
The auto backup feature is also turned on when you turn on automatic updates. This means that Appsmith is first backed up before updating to the latest version. If you see any errors during deployment, you can use the backup version to roll back. For more information, see [the rollback instructions using the `appsmithctl` command](/getting-started/setup/instance-management/appsmithctl#restore-appsmith-instance). If you continue to experience issues, reach out to the support team on the <a href="#!" onclick="Intercom('show')">chat widget</a> available on this page.
```

Author: I'll rewrite this.

---
description: Follow the steps to update Appsmith to the latest version.
---

# Update Appsmith
This page provides instructions to update Appsmith to the latest version.

## Prerequisites
Ensure that you complete the following steps before starting the update process.

* For backup and update tasks, you need at least 2 GB of free storage.

> * You need to create a backup of the Appsmith instance before a manual update. To create a backup, see the [Backup Appsmith Instance](/getting-started/setup/instance-management/appsmithctl#backup-appsmith-instance) section.

Reviewer 2: ```suggestion
* You need to create a backup of the Appsmith instance before a manual update. To create a backup, see the [Backup Appsmith Instance](/getting-started/setup/instance-management/appsmithctl#backup-appsmith-instance) section.
```
* It is highly recommended to create a backup of the Appsmith instance before a manual update. To create a backup, see the [Backup Appsmith Instance](/getting-started/setup/instance-management/appsmithctl#backup-appsmith-instance) section.

* If you are on a version earlier than v1.9.2, first upgrade to version v1.9.2. To upgrade, see the [Checkpoint version and upgrade](/getting-started/setup/instance-management#checkpoint-version-and-upgrades) section.

## Docker Compose
Follow the below steps on Docker or platforms that use `docker-compose`. For example, AWS AMI or DigitalOcean.  

You can update Appsmith in one of the following ways:

* [Manual update](#manual-update)
* [Automatic update](#automatic-update)

### Manual update
Go to the root directory of your installation and run:

> 

Reviewer 2: ```suggestion
:::info
If you do not remember the directory of you Appsmith installation, run:
`docker inspect -f '{{ (index .Mounts 0).Source }}’ <your-appsmith-container-id>`
::::
```

Author: The above command only gives the user the installation directory location on their machine, but this uses `docker` and not `docker-compose`.  Do we have a docker-compose equivalent as we are only using docker-compose commands in the documentation?

Author: > The above command only gives the user the installation directory location on their machine, but this uses `docker` and not `docker-compose`. Do we have a docker-compose equivalent, as we are only using docker-compose commands in the documentation? Also, this command is not returning the location for me. I'm getting this error:
> 
> ```
> "docker inspect" requires at least 1 argument.
> See 'docker inspect --help'.
> 
> Usage:  docker inspect [OPTIONS] NAME|ID [NAME|ID...]
> ```

Fixed it. The command had a backtick after the format string.  

`docker inspect -f '{{ (index .Mounts 0).Source }}' <your-appsmith-container-id>`

Reviewer 2: This command is applicable to docker-comose installations as well.

* If you are on a version earlier than v1.9.2, first upgrade to version v1.9.2. To upgrade, see the [Checkpoint version and upgrade](/getting-started/setup/instance-management#checkpoint-version-and-upgrades) section.

## Docker Compose
The below steps work on any platform that uses `docker-compose`. For example, Docker, AWS AMI, or DigitalOcean.

You can update Appsmith in one of the below ways:

* [Manual update](#manual-update)
* [Automatic update](#automatic-update)

:::info
If you don't know the installation directory, run:
```
docker inspect -f '{{ (index .Mounts 0).Source }}' <APPSMITH_CONTAINER_ID>
```
:::

### Manual update
Go to the root directory of your installation and run:

```
docker-compose pull && docker-compose rm -fsv appsmith && docker-compose up -d
```

### Automatic update

Follow the below steps to turn on the auto updates:

:::info
An automatic update first creates a backup and then starts the update.
:::

1. Go to the root directory of the Appsmith installation and run:

   ```
   docker-compose down
   ```

2. Open the `docker-compose.yml` file and uncomment the below code block:
   
   ```yaml

>    #auto_update:

Author: @sum35h - Please review if this block is correct.
